mysite/models/models.py

```diff
@@ -1415,10 +1415,24 @@ def _parse_w4_cooking_meals(self):
         # Meals contains 4 lists of lists. The first 3 are as long as the number of plates. The 4th is general shorter.
         emptyMeals = [emptyMeal for meal in range(4)]
         raw_meals_list = safe_loads(self.raw_data.get("Meals", emptyMeals))
+        # Make the sublists maxMeals long
         for sublistIndex, value in enumerate(raw_meals_list):
             if isinstance(raw_meals_list[sublistIndex], list):
                 while len(raw_meals_list[sublistIndex]) < maxMeals:
                     raw_meals_list[sublistIndex].append(0)
+                while len(raw_meals_list[sublistIndex]) > maxMeals:
+                    raw_meals_list[sublistIndex].pop()
+        
+        # Create meal dict
+        self.meals = {}
+        i = 0 
+        for lvl in raw_meals_list[0]:
```

Comment:
* Enumerating the list will get us a counter we don't have to manually modify.
`for mealIndex, mealLevel in enumerate(raw_meals_list[0]):`
mealIndex could be used in place of `i`, then the 2 lines creating and incrementing `i` can be removed. 
* This could be combined with the code block just below as well, since they are both iterating through raw_meals_list[0].